helper/schema/resource_data.go

```diff
@@ -604,6 +607,67 @@ func (d *ResourceData) GetRawConfig() cty.Value {
 	return cty.NullVal(schemaMap(d.schema).CoreConfigSchema().ImpliedType())
 }
 
+// GetRawConfigAt is a helper method for retrieving specific values
+// from the RawConfig returned from GetRawConfig. It returns the cty.Value
+// for a given cty.Path or an error diagnostic if the value at the given path does not exist.
+//
+// GetRawConfigAt is considered advanced functionality, and
+// familiarity with the Terraform protocol is suggested when using it.
+func (d *ResourceData) GetRawConfigAt(valPath cty.Path) (cty.Value, diag.Diagnostics) {
```

Comment:
I would check that this works for nested blocks; specifically traversing across `cty.SetVal` values is weird and I think `cty.Walk` is going to fail there.

Reply:
Good callout! I've added more test cases in commit [d1d4c14](https://github.com/hashicorp/terraform-plugin-sdk/pull/1375/commits/d1d4c14d859e91a7dbbd8bb12aad9cfadf0377c0). The Set and Set nested block traversal test cases are passing 👍🏾